### ISSUE TITLE
Keras ImageDataGenerator Validation_Split

### ISSUE THREAD

**Issue by broytburd** [keras-team/keras#14277]
How does the Keras "ImageDataGenerator" choose Validation data when using the parameter "validation_split"? 

I notice that a model I am training gets better validation performance when I randomly select the validation data manually using random.sample(), rather than with ImageDataGenerator.

And when I look into the images that "ImageDataGenerator" selects for validation, it seems to grab the data in random chunks, rather than over a uniform spread. This causes overfitting in my model. 

Here is some example code for what I am talking about:

`image_gen = tf.keras.preprocessing.image.ImageDataGenerator(validation_split = 0.2)`

`train_image_gen = image_gen.flow_from_directory(directory,`
                                              `target_size=new_shape,`
                                               `batch_size=batch_size,`
                                               `class_mode='categorical',`
                                               `subset = 'training')`

`val_image_gen = image_gen.flow_from_directory(directory,`
                                               `target_size=new_shape,`
                                               `batch_size=batch_size,`
                                               `class_mode='categorical',`
                                               `subset = 'validation')`

I understand this may not be an issue in a traditional sense, but is there a way to change the way ImageDataGenerator selects the images for the validation set?

**Comment by bebebib** [comment]
"I notice that a model I am training gets better validation performance when I randomly select the validation data manually using random.sample(), rather than with ImageDataGenerator."

^^This turned out to be a data issue on my part, but I do still wonder if there is a way to change how ImageDataGenerator selects validation data.
